<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDialog</class>
 <widget class="QDialog" name="SettingsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>577</width>
    <height>462</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>99</width>
    <height>99</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>664</width>
    <height>500</height>
   </size>
  </property>
  <property name="windowTitle">
   <string notr="true">Settings</string>
  </property>
  <property name="windowFilePath">
   <string notr="true"/>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Main</string>
      </attribute>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>231</width>
         <height>112</height>
        </rect>
       </property>
       <property name="title">
        <string>Compute</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_17">
          <item>
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>CPU threads</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="n_cpu">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>64</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>64</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <number>1024</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_18">
          <item>
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>GPU offload</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="n_gpu">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>64</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>64</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <number>9999</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>130</y>
         <width>231</width>
         <height>221</height>
        </rect>
       </property>
       <property name="title">
        <string>Generation</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Max generation</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="maxTok">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>64</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>64</width>
              <height>28</height>
             </size>
            </property>
            <property name="minimum">
             <number>-1</number>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Context length</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="conLen">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>64</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>64</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Seed</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="Seed">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>150</width>
              <height>28</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="eosToNL">
          <property name="text">
           <string>Treat EOS as NL</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="NLtoTurn">
          <property name="text">
           <string>Treat NL as end of turn</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Sampling</string>
      </attribute>
      <widget class="QGroupBox" name="groupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>161</width>
         <height>151</height>
        </rect>
       </property>
       <property name="title">
        <string>Sampling type</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QRadioButton" name="sampGreedy">
          <property name="text">
           <string>Greedy</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="sampTopP">
          <property name="text">
           <string>Top P</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="sampMiro1">
          <property name="text">
           <string>Mirostat v1</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="sampMiro2">
          <property name="text">
           <string>Mirostat v2</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>10</y>
         <width>187</width>
         <height>231</height>
        </rect>
       </property>
       <property name="title">
        <string>Top P / Top K</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_19">
          <item>
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>Top P</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="topP">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.050000000000000</double>
            </property>
            <property name="value">
             <double>0.950000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_20">
          <item>
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>Top K</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="topK">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="minimum">
             <double>0.000000000000000</double>
            </property>
            <property name="maximum">
             <double>1000000.000000000000000</double>
            </property>
            <property name="value">
             <double>40.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_21">
          <item>
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>Tail-free Z</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="tailFree">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_22">
          <item>
           <widget class="QLabel" name="label_21">
            <property name="text">
             <string>Min P</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="minP">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.010000000000000</double>
            </property>
            <property name="value">
             <double>0.050000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_23">
          <item>
           <widget class="QLabel" name="label_24">
            <property name="text">
             <string>Typical P</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="typP">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>1.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="geometry">
        <rect>
         <x>190</x>
         <y>250</y>
         <width>214</width>
         <height>112</height>
        </rect>
       </property>
       <property name="title">
        <string>Mirostat</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_24">
          <item>
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>Learning rate</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="learnRate">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <double>1.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.050000000000000</double>
            </property>
            <property name="value">
             <double>0.300000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_25">
          <item>
           <widget class="QLabel" name="label_23">
            <property name="text">
             <string>Entropy</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="Entropy">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <double>9999.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.500000000000000</double>
            </property>
            <property name="value">
             <double>3.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>250</y>
         <width>167</width>
         <height>112</height>
        </rect>
       </property>
       <property name="title">
        <string>Repeat penalty</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_26">
          <item>
           <widget class="QLabel" name="label_25">
            <property name="text">
             <string>Last N</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="lastN">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximum">
             <number>1000000</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_27">
          <item>
           <widget class="QLabel" name="label_26">
            <property name="text">
             <string>Penalty</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="repPenalty">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>75</width>
              <height>28</height>
             </size>
            </property>
            <property name="singleStep">
             <double>0.050000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="geometry">
        <rect>
         <x>390</x>
         <y>10</y>
         <width>151</width>
         <height>111</height>
        </rect>
       </property>
       <property name="title">
        <string>Temperature</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QDial" name="tempKnob">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>50</width>
            <height>50</height>
           </size>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>30</number>
          </property>
          <property name="notchesVisible">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="tempEdit">
          <property name="maximum">
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.050000000000000</double>
          </property>
          <property name="value">
           <double>0.300000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>UI</string>
      </attribute>
      <widget class="QComboBox" name="enterButn">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>531</width>
         <height>27</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>Enter sends text, Shift+Enter adds new line</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Enter adds new line, Shift+Enter sends text</string>
        </property>
       </item>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>181</width>
         <height>19</height>
        </rect>
       </property>
       <property name="text">
        <string>Enter button behavior</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="fixMD">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>341</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Automatically fix markdown in chat log</string>
       </property>
      </widget>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SettingsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
